pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -413,13 +419,130 @@ func newMigrationMonitor(vmi *v1.VirtualMachineInstance, l *LibvirtDomainManager
 		options:                  options,
 		migrationErr:             migrationErr,
 		progressWatermark:        int64(0),
+		remainingData:            int64(0),
 		progressTimeout:          options.ProgressTimeout,
 		acceptableCompletionTime: options.CompletionTimeoutPerGiB * getVMIMigrationDataSize(vmi),
 	}
 
 	return monitor
 }
 
+func (m *migrationMonitor) shouldTriggerTimeout(elapsed int64, domSpec *api.DomainSpec) bool {
+	if m.acceptableCompletionTime == 0 {
+		return false
+	}
+
+	if domSpec.Metadata.KubeVirt.Migration != nil && domSpec.Metadata.KubeVirt.Migration.Mode == v1.MigrationPostCopy {
+		return false
+	}
+
+	if elapsed > m.acceptableCompletionTime {
+		return true
+	}
+
+	return false
+}
+
+func (m *migrationMonitor) shouldTriggerPostCopy(elapsed int64, domSpec *api.DomainSpec) bool {
+	if m.shouldTriggerTimeout(elapsed, domSpec) && m.options.AllowPostCopy {
+
+		return true
+	}
+	return false
+}
+
+func (m *migrationMonitor) isMigrationProgressing(domainSpec *api.DomainSpec) bool {
+	logger := log.Log.Object(m.vmi)
+
+	now := time.Now().UTC().Unix()
+
+	// If the migration is in post copy mode we should
+	// not abort the job as the migration would lose state
+	// so post copy can not timeout
+	if domainSpec.Metadata.KubeVirt.Migration != nil && domainSpec.Metadata.KubeVirt.Migration.Mode == v1.MigrationPostCopy {
+		return true
+	}
```

Comment:
Seems like this belongs to a different question that should be under its own method in parallel with the others switch/if-else (and asked before `isMigrationProgressing`).

Something like `isPostCopyEnabled` ?

Reply:
makes sense. i changed this